### PR TITLE
borrowed $ to make possix.Time string friendly

### PR DESCRIPTION
**Pull request by Zectbumo**
now possix.Time can be easily used with `echo` and strformat etc.

### CODE REVIEW

File: lib/pure/uri.nim [hunk 1]
```diff
@@ -123,13 +123,13 @@ func decodeUrl*(s: string, decodePlus = true): string =
   setLen(result, j)
 
 func encodeQuery*(query: openArray[(string, string)], usePlus = true,
-    omitEq = true): string =
+    omitEq = true, sep = '&'): string =
```
Comment (ghost) [inline lib/pure/uri.nim]: does this belong to this PR?

Comment (Zectbumo) [reply lib/pure/uri.nim]: no this is another PR I'm working on. first PR should be from [b699c1a](https://github.com/nim-lang/Nim/pull/19812/commits/b699c1a44c120a24fff0436f0215ca39e64d0963)

Comment (ghost) [reply lib/pure/uri.nim]: @Zectbumo you should then only make this PR change the posix stuff and not touch the uri stuff

**Comment by Araq** [comment]
But what string representation does it produce? Some junk number? posix.nim is not for convenience anyway, it's a low level wrapper.

**Comment by Zectbumo** [comment]
This may be too low level. As it stands, the Time type has other convenience operators that it borrowed such as the `-` for calculating time deltas and `==` for equality. I thought that the `$` would make a nice addition to the borrowed operator set for easy displaying:
`echo time2 - time1, "sec delta"`

**Comment by Araq** [comment]
Sorry, but rejected for the time being.

**Pull request closed without merging**